metrics/web/docs/features/router_widget/router_widget.md

```diff
@@ -0,0 +1,147 @@
+
+Moreover, this layer contains the  `MetricsRouterDelegate` class - a delegate that is used by the `Router` widget to build and configure a navigating widget. 
+
+The `build` method in a `MetricsRouterDelegate` needs to return a `Navigator`. In this way, our navigation changes will be reflected in the address bar.
+
+The `onPopPage` callback uses `notifyListeners`. When the `MetricsRouterDelegate` notifies its listeners, the `Router` widget is likewise notified that the `MetricsRouterDelegate`'s `currentConfiguration` has changed and that it's the `build` method needs to be called again to build a new `Navigator`.
+
+Example:
+```
+class MetricsRouterDelegate extends RouterDelegate<String>
+    with ChangeNotifier, PopNavigatorRouterDelegateMixin<String> {
+  final _navigatorKey = GlobalKey<NavigatorState>();
+  final List<Page> _pages = [];
+  final bool isLoggedIn;
+
+  MetricsRouterDelegate({this.isLoggedIn});
+
+  @override
+  Widget build(BuildContext context) {
+    return Navigator(
+      key: _navigatorKey,
+      pages: _pages,
+      onPopPage: (route, result) {
+        if (!route.didPop(result) || !isLoggedIn) {
+          return false;
+        }
+        notifyListeners();
+        return true;
+      },
+    );
+  }
+
+  @override
+  GlobalKey<NavigatorState> get navigatorKey => _navigatorKey;
+
+  @override
+  Future<void> setNewRoutePath(String path) async {
+    if (path == RouteName.dashboard) {
+      _pages.add(
+        PageGenerator.generatePage(
+          name: RouteName.dashboard,
+          isLoggedIn: isLoggedIn,
+        ),
+      );
+    }
+    notifyListeners();
+    return;
+  }
+}
+
+
+```
+
+The structure of the presentation layer shown in the following diagram: 
+
+![Presentation layer diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/router_widget/metrics/web/docs/features/router_widget/diagrams/router_widget_presentation.puml)
+
+When the RouteInformationProvider emits a new route, it notifies the Router about changes.
+
+Router delegates parsing of changes to the RouteInformationParser, which converts it into an abstract data type T.
+
+RouterDelegate’s setNewRoutePath method is called with this data type and must update the application state to reflect the change and call notifyListeners.
+
+When notifyListeners is called, it tells the Router to rebuild the RouterDelegate (using its build() method).
+
+RouterDelegate.build() returns a new Navigator, whose pages now reflect the change to the app state.
+
+The following sequence diagram shows the process of changing the route: 
+![Route change diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/router_widget/metrics/web/docs/features/router_widget/diagrams/route_change_sequence.puml)
+
```

Comment:
Let's add the Testing section as follows: 

```suggestion
## Testing
```

Here, describe how the existing test codebase is affected by the changes with Router class. What changes required to make tests work properly with a new navigation system? And so on.